sycl/test/inline-asm/asm_16_empty.cpp

```diff
@@ -0,0 +1,73 @@
+// UNSUPPORTED: cuda
+// REQUIRES: gpu,linux
+// RUN: %clangxx -fsycl %s -DINLINE_ASM -o %t.out
+// RUN: %t.out
+
+#include "include/asmcheck.h"
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <string>
+#include <vector>
+
+constexpr int LIST_SIZE = 1024;
+using arr_t = std::vector<cl::sycl::cl_int>;
+constexpr auto sycl_write = cl::sycl::access::mode::write;
+
+// class is used for kernel name
+template <typename T>
+class no_opts;
+
+template <typename T>
+void process_buffers(cl::sycl::queue &deviceQueue, T *pc, size_t sz) {
```

Comment:
What is the point of having template parameter here? I see only on call of this function(with int as T).